target_snowflake/__init__.py

```diff
@@ -431,13 +431,64 @@ def load_stream_batch(stream, records, row_count, db_sync, no_compression=False,
         # reset row count for the current stream
         row_count[stream] = 0
 
+# create schema matching record (deleting unwanted keys)
+def schema_from_record(record, schema):
+    unwanted = set(schema) - set(record)
+
+    for unwanted_key in unwanted:
+        del schema[unwanted_key]
+
+    return schema
+
+
+# group records by unique combination of keys of a record
+def group_records(records, full_schema):
+    grouped_records = {}
+    
+    for record_key, record in records.items():
+      key = "-".join((list(record.keys())))
+      schema = schema_from_record(record, full_schema)
+
+      if key in grouped_records:
+        grouped_records[key]["records"][record_key] = record
+      else:
+        grouped_record[key] = {"records": {record_key: record}, "schema": schema})
```

Comment:
Don't know if Python supports it, but you can put the record keys in a `Set` object and use the set as the key of the grouped_records dict. That way you don't have to worry about, for example, the order of the keys of the record!

Reply:
A non-hashable object can not be a key. But I can  try `frozenset`.